lib/bambora/json_request.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+module Bambora
+  class JSONRequest
+    attr_reader :client
+
+    def initialize(client)
+      @client = client
+      @path = ''
+    end
+
+    def request(method:, path:, params: {}, body: {})
+      parse_response(
+        client.request(method: method, path: path, params: params, body: body.to_json.to_s),
+      )
+    end
+
+    private
+
+    def parse_response(resp)
+      symbolize_keys(JSON.parse(resp.body))
+    end
+
+    def symbolize_keys(obj)
+      return obj unless obj.is_a?(Hash)
```

Comment:
what if you get a hash which holds a collection of hashes e.g. 
```ruby
obj = {
  'user' => {
    'accounts' => [
      {
        'number' => '123',
      },
      {
        'number' => '456',
      },
    ]
  }
}
def symbolize_keys(obj)
  return obj unless obj.is_a?(Hash)

  obj.each_with_object({}) do |(k, v), hash|
    hash[k.to_sym] = symbolize_keys(v)
    hash
  end
end

symbolize_keys(obj) => {:user=>{:accounts=>[{"number"=>"123"}, {"number"=>"456"}]}}
```

Reply:
took from rails source code:
```ruby
def _deep_transform_keys_in_object(object, &block)
      case object
      when Hash
        object.each_with_object({}) do |(key, value), result|
          result[yield(key)] = _deep_transform_keys_in_object(value, &block)
        end
      when Array
        object.map { |e| _deep_transform_keys_in_object(e, &block) }
      else
        object
      end
    end

    _deep_transform_keys_in_object(obj){ |key| key.to_sym } => {:USER=>{:ACCOUNTS=>[{:NUMBER=>"123"}, {:NUMBER=>"456"}]}}
```
You could wrap that method into your method

Reply:
Nice, thanks @bilogub!